webapp/src/components/AssetTransactionHistory/AssetTransactionHistory.js

```diff
@@ -7,25 +7,47 @@ import { locations } from 'locations'
 import AddressBlock from 'components/AddressBlock'
 import BlockDate from 'components/BlockDate'
 import Mana from 'components/Mana'
-import { assetType, publicationType } from 'components/types'
+import { assetType, publicationType, bidType } from 'components/types'
 import { t } from '@dapps/modules/translation/utils'
-import { LISTING_STATUS } from 'shared/listing'
+import { LISTING_STATUS, LISTING_TYPES } from 'shared/listing'
+import { shortenOwner } from 'shared/map'
 import { findAssetPublications } from 'shared/publication'
-import { distanceInWordsToNow, shortenAddress } from 'lib/utils'
+import { distanceInWordsToNow } from 'lib/utils'
 
 import './AssetTransactionHistory.css'
 
 export default class AssetTransactionHistory extends React.PureComponent {
   static propTypes = {
     asset: assetType.isRequired,
-    publications: PropTypes.objectOf(publicationType)
+    publications: PropTypes.objectOf(publicationType),
+    bids: PropTypes.arrayOf(bidType)
   }
 
-  getAssetPublications() {
-    const { asset, publications } = this.props
-    return findAssetPublications(publications, asset, LISTING_STATUS.sold).sort(
-      (a, b) => (a.block_number > b.block_number ? -1 : 1)
+  componentWillMount() {
+    this.props.onFetchAssetTransactionHistory()
+  }
+
+  getAssetListings() {
+    const { asset, publications, bids } = this.props
+    const assetPublications = findAssetPublications(
+      publications,
+      asset,
+      LISTING_STATUS.sold
     )
+    return [...assetPublications, ...bids]
+      .map(listing => ({
+        block_number: listing.block_number,
+        price: listing.price,
+        block_time_created_at: listing.block_time_created_at,
+        block_time_updated_at: listing.block_time_updated_at,
+        id: listing.tx_hash || listing.id,
+        from: listing.owner || listing.seller,
+        to: listing.buyer || listing.bidder,
+        type: listing.tx_hash ? LISTING_TYPES.PUBLICATION : LISTING_TYPES.BID
+      }))
```

Comment:
The only way of telling if a listing is a bid or a publication is via the tx_hash prop? It scares me a bit because tomorrow we can add a tx_hash prop to a bid and this would break.

Don't you think it's a good idea to at least make this two separate methods to ease maintainability?

Something _like_ this:

`.map(listing => listing.tx_hash ? this.buildPublication(listing) : this.buildBid(listing))`